pilot/pkg/model/push_context.go

```diff
@@ -45,6 +47,9 @@ import (
 	"istio.io/pkg/monitoring"
 )
 
+// TODO: make this configurable
+const cacheSize = 200
```

Comment:
I don't think we need a full blown LRU cache within PushContext. PushContext _is_ already a cache. The problem is we are storing an intermediate state (LoggingConfig), and then for every listener (often times called 1000s of times), we convert LoggingConfig -> Envoy config.

I think simply storing the computed config in LoggingConfig would be sufficient optimization

Reply:
Or something like `computedMetricsFilters` would provide cross-Proxy caching, although that may not really be super important. The above change would already give us orders of magnitude improvement (O(listeners))

Reply:
Storing in LoggingConfig would require the caller to only fetch a single LoggingConfig though, access log is currently not structured like that. So maybe not the right path.

I would say to just follow the same pattern as `computedMetricsFilters`. Its pretty similar to what you have here except using a simple map vs LRU, and storing in Telemetries

Reply:
> Just 1 single telemetry. 5000 proxies. Access logs config are EXTREMELY expensive to generate since its so verbose and so many fields (https://github.com/envoyproxy/envoy/issues/21718)

In this scenario, I believe most of costs are in `setTCPAccessLog\setHTTPAccessLog`(I do not have a real pprof)
If we just cache LoggingConfig, may not help.

> Its pretty similar to what you have here except using a simple map vs LRU, and storing in Telemetries
> On a fairly small cluster I end up with 400 access log configs. The LDS is 700kb. Disabling access logs drops it down to 200kb.

There's a huge memory cost if we use a simple map? 500kb * mum of proxy?

Reply:
We already do it for http and tcp filters for stats. Almost all proxies have the same config so it will end up being pretty small is my guess?

Reply:
if it is an issue we can cap the map size, but lru is never needed IMO since the scope is to a single PushContext which is already short lived

Reply:
updated the implements.